rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+RTR will use the trip IDs from the trainsheet edit events when determining what trip ID to publish for a vehicle matching that trip. The trip matching logic itself will still live with RTR at this point, but if RTR believes that a train is a particular trip from the trainsheet, it SHOULD use the corresponding trip ID in the public GTFS-rt feed. However, if a train that is assigned to a scheduled trip from Glides changes state in such a way that it should be assigned to a different trip by RTR's logic (for instance: unexpectedly going off its current pattern, or changing directions), RTR MAY assign that train to a different trip of its choosing. RTR will still be free to create its own added trips as needed.
+
+Additionally, RTR will need to handle cases where current service as reflected in GTFS static does not match the HASTUS-based trainsheets in Glides, due to planned disruptions modeled by `gtfs_creator`. In these cases, the data will manifest as trainsheet update messages with HASTUS-based trip IDs that do not match up with the trip IDs currently running per the GTFS static schedule. This will continue to be a potential issue, even with trip matching handled in Glides, so it is explored in more depth in the [Reference-level explanation](#reference-level-explanation) section.
+
+This intermediate state is proposed to allow for a more incremental implementation that's easier to deliver and provides some value without requiring too much deep refactoring of RTR. It can optionally be skipped if the Glides and Transit Data teams collectively decide that it creates more net work for not enough additional benefit.
+
+## Train - trip assignment feed
+
+In a future state, Glides will be responsible for light rail trip assignments, and generate a new event feed of train-to-trip assignment events. RTR will use this as its primary source of information 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## GTFS and TODS schedule data
+
+Light rail schedule data in GTFS and (eventually) TODS files will be produced by `gtfs_creator` based on an export from HASTUS. The HASTUS trips for the lead car will be the basis for GTFS trips, and GTFS trips will take on their IDs. Integrating the schedule for multi-car trips (that is, trips with a trailer) into the TODS data will be defined in a separate discussion about the adoption of TODS. When `gtfs_creator` takes a trip from HASTUS and outputs it with only minor factual corrections, it SHOULD retain the original trip ID. However, if a trip is actively modified as part of the `gtfs_creator` disruption modeling process, its ID MUST be changed so as to not overlap with the HASTUS trip IDs.
```

Comment:
Just a commentary on current state of GTFS creator modifying things. We've got a few different relevant cases:

- HASTUS corrections: These are generally minor edits (moving a stop, changing a time) and are handled by patches that don't change the trip_id
- HASTUS modeled diversions: Most diversions anymore we are receiving a new HASTUS export for. These come with a different (HASTUS assigned) service_id and trip_id from what's in the rating. We don't change these trip_ids/service_ids from what HASTUS has provided.
- GTFS creator-modeled diversions: This will update the service_ids and trip_ids in semi-predictable ways. They will never match the original HASTUS trip. That said, these will sometimes update trip_ids for trips which were checked, but which the diversion did not apply to. Example: if a diversion affects only one branch of the RL we will update the trip_ids for both branches.
- Things that are modeled as diversions in GTFS creator even though they are not: These will update the service_id/trip_ids even though the underlying trip is not modified. Example: we will sometimes run a shuttle at the end of a train trip as an ExtendedShuttle. This will modify the train's trip_id/service_id even though the trip isn't actually modified, as an implementation detail. These are usually combined with HASTUS based diversions (truncate the orange line with a new HASTUS export, and add a shuttle at the end).

Reply:
Thanks, this is all helpful context.

One thing to keep in mind is that this is all just for light rail, so while it's not a bad idea to keep the Red Line or Orange Line, we're specifically concerned about the Green and Mattapan Lines here, which I mention just in case there is a marked difference in terms of how often Green Line disruptions are modeled in a particular way versus how frequently they're modeled that way on the heavy rail.

Just curious, regarding the last point, do you have a sense of how important an implementation detail that is? As in, is it something that could easily be changed in the future? If not then that's fine, just trying to get a sense of which cases I should plan for existing in the long term.

Reply:
It's definitely not an important detail, and in fact caused us some mild heartburn recently. It's not exactly an easy change, but that has more to do with how gtfs_creator is structured than anything else. I don't see a problem with changing it for this effort.